apis/definition.yaml

```diff
@@ -0,0 +1,83 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: CompositeResourceDefinition
+metadata:
+  name: xsecuritygroups.aws.platform.upbound.io
+spec:
+  group: aws.platform.upbound.io
+  names:
+    kind: XSecurityGroup
+    plural: xsecuritygroups
+  claimNames:
+    kind: SecurityGroup
+    plural: securitygroups
+  versions:
+    - name: v1alpha1
+      served: true
+      referenceable: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                parameters:
+                  type: object
+                  description: Parameters used to customize the cluster
+                  properties:
+                    id:
+                      type: string
+                      description: ID of this Network that other objects will use to refer to it.
+                    region:
+                      type: string
+                      description: Region is the region you'd like your resource to be created in.
+                    deletionPolicy:
+                      description: Delete the external resources when the Claim/XR is deleted. Defaults to Delete
+                      enum:
+                        - Delete
+                        - Orphan
+                      type: string
+                      default: Delete
+                    providerConfigName:
+                      description: Crossplane ProviderConfig to use for provisioning this resources
+                      type: string
+                      default: default
+                    rules:
+                      type: array
+                      items:
+                        type: object
+                        properties:
+                          type:
+                            type: string
+                            description: Type of rule, ingress (inbound) or egress (outbound)..
+                            enum:
+                              - ingress
+                              - egress
+                          fromPort:
+                            type: integer
+                          toPort:
+                            type: integer
+                          description:
+                            type: string
+                            default: Crossplane generated Security Group Rule
+                          protocol:
+                            type: string
+                          sourceSecurityGroupName:
+                            type: string
+                            description: Reference to a SecurityGroup to populate sourceSecurityGroupName
```

Comment:
```suggestion
                            description: Reference to a SecurityGroup to populate sourceSecurityGroupName (optional).
```

Reply:
we have CEL rules now - so i will not optional in description text